File: java/Sieve.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.Scanner;

public class Sieve {
    private static List<Integer> sieve(int n) {
        boolean[] nums = new boolean[n+1];
        Arrays.fill(nums, true);
        
        for (int i = 3; n >= i * i; i = i + 2) {
            if (!nums[i]) continue;
            for (int j = i * i; j <= n; j = j + i * 2) {
                if (nums[j]) nums[j] = false;
            }
        }

        List<Integer> primes = new ArrayList<>();
        primes.add(2);
        for (int i = 3; i <= n; i = i + 2) {
            if (nums[i]) primes.add(i);
        }

        return primes;
    }
    
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = Integer.parseInt(sc.next());
        
        List<Integer> primes = sieve(n);

        System.out.println( primes.size() + " " + primes.get(primes.size() - 1) );
    }
}

File: go/sieve.go
package main

import (
	"fmt"
)

func sieve(n int) []int {
	nums := make([]bool, n + 1)
	for i := 3; i <= n; i = i + 2 {
		nums[i] = true
	}

	for i := 3; n >= i * i; i = i + 2 {
		if !nums[i] {
			continue
		}

		for j := i * 3; j <= n; j = j + i * 2 {
			if nums[j] {
				nums[j] = false
			}
		}
	}

	primes := []int{2}
	for i := 3; i <= n; i = i + 2 {
		if nums[i] {
			primes = append(primes, i)
		}
	}
	
	return primes
}

func main() {
	var n int
	fmt.Scan(&n)
	
	primes := sieve(n)
	fmt.Println(len(primes), primes[len(primes)-1])
}
